src/jdk.jfr/share/classes/jdk/jfr/internal/tool/Metadata.java

```diff
@@ -127,16 +162,76 @@ public void execute(Deque<String> options) throws UserSyntaxException, UserDataE
         try (PrintWriter pw = new PrintWriter(System.out)) {
             PrettyWriter prettyWriter = new PrettyWriter(pw);
             prettyWriter.setShowIds(showIds);
-            try (RecordingFile rf = new RecordingFile(file)) {
-                List<Type> types = PRIVATE_ACCESS.readTypes(rf);
-                Collections.sort(types, new TypeComparator());
-                for (Type type : types) {
-                    prettyWriter.printType(type);
+
+            List<Type> types = null;
+            if (file != null) {
+                // has recording.jfr, read metadata from file
+                try (RecordingFile rf = new RecordingFile(file)) {
+                    types = JdkJfrConsumer.instance().readTypes(rf);
+                    Collections.sort(types, new TypeComparator());
                 }
+            } else {
+                // don't have recoring.jfr, read metadata from metadata.bin
+                types = TypeLibrary.getInstance().getTypes();
+            }
+
+            for (Type type : types) {
+                if (!(type instanceof PlatformEventType)) {
+                    continue;
+                }
+                if (foundEventFilter) {
+                    if (acceptedEvents.contains(type.getName())) {
```

Comment:
I would expect filtering for the metadata command to work like the 'print' command (accepting acronyms, wildcards etc.)

Reply:
Added.